### PR TITLE
[ESLint] Lint test files

### PR DESCRIPTION
**Pull request by bbecquet**
## Description
Apply ESLint to test suite files, with slightly different rules:
 - jest+node as env to manage jest-specifics features without override (`describe`, `test`, etc.)
 - I set the `max-len` rule to `warn` only, because tests are full of loooong selectors and I don't think it's crucial. What do you think?

## Why
No reason to have different coding practices on test code.

### CODE REVIEW

File: tests/integration/helpers/response_handler.js [hunk 1]
```diff
@@ -1,42 +1,42 @@
...
+    const responseHandler = new ResponseHandler(page);
+    await responseHandler.prepareResponse();
+    return responseHandler;
   }
 
   addPreparedResponse(response, query, options) {
-    let alreadySetResponse = this.preparedResponses.find((preparedResponse) => preparedResponse.query.toString() === query.toString())
-    if(!alreadySetResponse) {
-      this.preparedResponses.push({response, query, options})
+    const alreadySetResponse = this.preparedResponses.find(preparedResponse => preparedResponse.query.toString() === query.toString());
```
Comment (GuillaumeGomez) [inline tests/integration/helpers/response_handler.js]: I assume this line is above 100 columns?

File: tests/integration/helpers/response_handler.js [hunk 2]
```diff
@@ -1,42 +1,42 @@
...
+        interceptedRequest.headers['Access-Control-Allow-Origin'] = '*';
+        let status = 200;
+        if (preparedResponse.options) {
+          if (preparedResponse.options.status) {
+            status = preparedResponse.options.status;
           }
         }
-        await interceptedRequest.respond({status : status, body : JSON.stringify(preparedResponse.response), headers  : interceptedRequest.headers})
-        return
+        await interceptedRequest.respond({status: status, body: JSON.stringify(preparedResponse.response), headers: interceptedRequest.headers});
```
Comment (GuillaumeGomez) [inline tests/integration/helpers/response_handler.js]: I assume this line is above 100 columns?

File: tests/integration/server_start.js [hunk 3]
```diff
@@ -1,54 +1,55 @@
...
-const nock = require('nock')
-
-let {...poiNoName} = require('../__data__/poi.json')
-  /* default test with matching name & local_name */
-  poiNoName.id = 'osm:way:453204'
-  nock(/idunn_test\.test/)
-    .persist(true)
-    .get(/osm:way:453204/)
-    .reply(200, JSON.stringify(poiNoName))
+/* eslint-disable no-console */
```
Comment (GuillaumeGomez) [inline tests/integration/server_start.js]: Can you put it directly above the line where `console` is used instead please?

File: tests/integration/tests/autocomplete.js [hunk 4]
```diff
@@ -1,303 +1,306 @@
...
-    'accept-language': 'de,en;q=0.8'
-  })
-  const responseHandler = await ResponseHandler.init(langPage)
-  const autocompleteHelper = new AutocompleteHelper(langPage)
+/* globals Poi */
+import {clearStore, initBrowser, wait} from '../tools';
+import AutocompleteHelper from '../helpers/autocomplete';
+import ResponseHandler from '../helpers/response_handler';
+const configBuilder = require('@qwant/nconf-builder');
+const config = configBuilder.get();
```
Comment (GuillaumeGomez) [inline tests/integration/tests/autocomplete.js]: We should maybe enforce capitalized global consts. What do you think about it?

File: tests/integration/tests/autocomplete.js [hunk 5]
```diff
@@ -1,303 +1,306 @@
...
+const SUGGEST_MAX_ITEMS = config.services.geocoder.max_items;
+
+let browser;
+let page;
+let autocompleteHelper;
+let responseHandler;
+const mockAutocomplete = require('../../__data__/autocomplete.json');
+const mockAutocompleteAllTypes = require('../../__data__/autocomplete_type.json');
+
+beforeAll(async() => {
```
Comment (GuillaumeGomez) [inline tests/integration/tests/autocomplete.js]: No more whitespace after `async`?

File: tests/integration/tests/server.js [hunk 6]
```diff
@@ -1,39 +1,38 @@
...
+    .expect(204, done);
 });
 
 test('responds to events and update metrics', done => {
   server
     .get('/metrics')
     .expect(200)
     .then(response => {
-      let currentFavSaveCount = parseInt(response.text.match(/\nerdapfel_favorite_save_count (\d*)/)[1])
+      const currentFavSaveCount = parseInt(response.text.match(/\nerdapfel_favorite_save_count (\d*)/)[1]);
```
Comment (GuillaumeGomez) [inline tests/integration/tests/server.js]: I assume this line is above 100 columns?

File: tests/integration/tests/server.js [hunk 7]
```diff
@@ -42,26 +41,26 @@ test('responds to events and update metrics', done => {
           server.get('/metrics')
             .expect(200)
             .then(response => {
-              let newSaveCount = parseInt(response.text.match(/\nerdapfel_favorite_save_count (\d*)/)[1])
-              expect(newSaveCount).toBeGreaterThan(currentFavSaveCount)
+              const newSaveCount = parseInt(response.text.match(/\nerdapfel_favorite_save_count (\d*)/)[1]);
```
Comment (GuillaumeGomez) [inline tests/integration/tests/server.js]: I assume this line is above 100 columns?

File: .eslintrc.json [hunk 8]
```diff
@@ -77,5 +77,19 @@
...
+        {
+            "files": [
+                "tests/**/*.js"
+            ],
+            "env": {
+                "jest": true,
+                "node": true
+            },
+            "rules": {
+                "max-len": ["warn", { "code": 100, "tabWidth": 2 }]
```
Comment (GuillaumeGomez) [inline .eslintrc.json]: I'm still not a big fan of this. (I wanted to say it once again°

File: tests/integration/tests/autocomplete.js (hunk 4 above)
Comment (bbecquet) [reply tests/integration/tests/autocomplete.js]: Maybe, but in another PR :) There are a lot of things to improve on globals everywhere in the code, like stop polluting the `window` namespace. BTW, here this isn't really a global.

File: tests/integration/tests/autocomplete.js (hunk 5 above)
Comment (bbecquet) [reply tests/integration/tests/autocomplete.js]: This is a result of `eslint --fix`, and it doesn't seem to be invalid code. But I agree this is unclear. I'll change.

File: .eslintrc.json (hunk 8 above)
Comment (bbecquet) [reply .eslintrc.json]: Ok, I'll revert to error :)

**Pull request merged**